snap-rcp/src/main/java/org/esa/snap/rcp/status/PixelPosStatusLineElementProvider.java

```diff
@@ -144,7 +165,11 @@ public void pixelPosChanged(ImageLayer imageLayer,
                 scaleStr = "1:" + ((int) v == v ? (int) v : v);
             }
             zoomLevelLabel.setText(String.format(ZOOM_LEVEL_FORMAT, scaleStr, currentLevel));
-
+            double scale= rasterDataNode.getMultiLevelModel().getScale(currentLevel);
```

Comment:
This calculation only works for data reprojected on a map, like, sentinel-2.
But for Sentinel-3 L1 data it shows 1m, but actually it has a resolution of 300m at nadir.
For calculating the distance between two points the SphericalDistance class can be used. But the radian result still needs to be converted to meter.
Alternatively it can be extracted from the [RangeFinder](https://github.com/senbox-org/snap-desktop/blob/07bbb2cea825dc3de47fa51244a5fa5551dd2f61/snap-rcp/src/main/java/org/esa/snap/rcp/actions/interactors/RangeFinderInteractor.java#L196) code.
But probably best would be to use [RasterUtils.computeResolutionInKm()](https://github.com/senbox-org/snap-engine/blob/4a0e6f351e5395b4ce58067ee28830255125e323/snap-core/src/main/java/org/esa/snap/core/dataio/geocoding/util/RasterUtils.java#L249) and convert the unit to metre.

Reply:
Is the computeResolutionInKm method adapted for the reprojected data or It have to separate the different cases?

Reply:
I think yes. It should be usable for all cases. But it might be good to consider projected data differently. The result of this method might be 0.0102 for a S2 10 meter band or 0.00999. Because the calculation can't be absolute correct. So users might wonder why 10.2 metres are shown when it should actually be 10m.

Reply:
In the  RasterUtils.computeResolutionInKm() method, the resolution is the median value between all pixel pos. 
What's it better? Median value of all pixel pos or just median value of adjacent pixel around the current pixel pos of the mousse event.

Reply:
Maybe you just use `final SphericalDistance spherDist = new SphericalDistance(Lon, Lat)` and convert this into metre.
So that, RasterUtils.computeResolutionInKm() only serves as a template.

I think it will confuse users when the spacing is changing with the cursor movement, even it is correct.
An option could be, to compute only the spacing at the centre of the scene and make it constant in the status bar. Alternatively an option could be introduced to let users decide what shall be shown.

Reply:
How to know if data is resampled and reprojected?

Reply:
You can check the GeoCoding of the product/band. If it is an instance of CrsGeoCoding, then it is reprojected.
For resampled data there is now check possible.

Reply:
After resampling S3 L1 product, the resolution getting with spherical distance of the latitude geocoding are the same without resampling. The geocoding isn't modified by the resamping. It 's a bug?
How to retrieve lat and lon between 2 pixels on resampled image?

Furthermore, there are something wrong with geoCoding.getImageToMapTransform() ```geoCoding.getImageToMapTransform().getSourceDimensions()```=width of target image
```geoCoding.getImageToMapTransform().getTargetDimensions()```=height of target image.

Reply:
The GeoCoding stays the same. It just uses the different lat/lon tie-points or lat/lon bands.
But the resolution should actually change.

How to retrieve lat and lon between 2 pixels on resampled image?
As on any other image. 
```
GeoPos gp1 =GeoCoding.getGeoPos(new PixelPos(x1,y1), null);
GeoPos gp2 =GeoCoding.getGeoPos(new PixelPos(x2,y2), null);
DistanceMeasure measure = new SphericalDistance(gp1.lon, gp1.lat);
distance = measure.distance(gp2.lon, gp2.lat); // still in radian
```

On which implementation do you see the wrong Dimensions? Most GeoCoding implementaion rely on the default implementation in AbstractGeoCoding and there we use GeoTools to retrieve the transform.
`image2Map = CRS.findMathTransform(imageCRS, mapCRS);`
Either there is something wrong or already the definition of the imageCRS or the mapCRS is not correct.

Reply:
I try to look at this tomorrow. Rest of the day is already booked.